File: problems/2660-determine-the-winner-of-a-bowling-game/solution.cpp
class Solution {
public:
    int isWinner(vector<int>& player1, vector<int>& player2) {
        int score1 = 0;
        int score2 = 0;
        for (int i = 0; i < player1.size(); ++i) {
            int val1 = player1[i];
            if (i > 1 && (player1[i - 1] == 10 || player1[i - 2] == 10)) {
                val1 *= 2;
            }
            score1 += val1;
            int val2 = player2[i];
            if (i > 1 && (player2[i - 1] == 10 || player2[i - 2] == 10)) {
                val2 *= 2;
            }
            score2 += val2;
        }
        if (score1 > score2) {
            return 1;
        } else if (score2 > score1) {
            return 2;
        } else {
            return 0;
        }
    }
};

File: problems/2660-determine-the-winner-of-a-bowling-game/solution.py
class Solution:
    def isWinner(self, player1: List[int], player2: List[int]) -> int:
        def calculate_score(player):
            score = 0
            for i in range(len(player)):
                val = player[i]
                if i > 1 and (player[i-1] == 10 or player[i-2] == 10):
                    score += 2 * val
                else:
                    score += val
            return score
        
        score1 = calculate_score(player1)
        score2 = calculate_score(player2)
        
        if score1 > score2:
            return 1
        elif score2 > score1:
            return 2
        else:
            return 0